rules/exclude_files.bzl

```diff
@@ -0,0 +1,29 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+def _exclude_files_impl(ctx):
+    out = []
+    for src in ctx.files.srcs:
+        include = True
+        for suffix in ctx.attr.exclude_suffix:
+            if src.path.endswith(suffix):
+                include = False
+                break
+        if include:
+            out.append(src)
+    return [DefaultInfo(files = depset(out))]
+
+exclude_files = rule(
```

Comment:
Quick question: it looks like the [`pkg_files` rule](https://bazelbuild.github.io/rules_pkg/0.7.0/reference.html#pkg_files) has an `excludes` attribute that allows you to specify files to exclude from the pkg. Is there a reason to have a custom `exclude_files` rule?

Reply:
The `excludes` attribute in pkg_files needs to be either a label or filename and may not be a glob or other expression.

The CcSideProducts produce some combination of preprocessor outputs, LLVM-IR and assembly outputs and there is some file-naming shenanigans in the side products rules to ensure that we don't produce same-named outputs when certain files are compiled twice.

The `exclude_files` rule is specifically designed to allow filtering on file suffixes rather than exact filenames.